src/Stripe.net/Infrastructure/Public/StripeRequest.cs

```diff
@@ -99,6 +98,33 @@ public override string ToString()
             return new Uri(b.ToString());
         }
 
+        private static AuthenticationHeaderValue BuildAuthorizationHeader(
+            RequestOptions requestOptions)
+        {
+            string apiKey = requestOptions?.ApiKey ?? StripeConfiguration.ApiKey;
+
+            if (string.IsNullOrEmpty(apiKey))
+            {
+                var message = "No API key provided. Set your API key using "
+                    + "`StripeConfiguration.ApiKey = \"<API-KEY>\"`. You can generate API keys "
```

Comment:
Should this mention `requestOptions` as an alternative if it's not null?

Reply:
I don't want to make the error message too long. The docs at https://stripe.com/docs/api/authentication do show how to set the API key on a per-request basis.